netty/src/main/java/io/grpc/transport/netty/Utils.java

```diff
@@ -91,10 +92,13 @@ static ByteBuf toByteBuf(ByteBufAllocator alloc, ByteBuffer source) {
   public static Metadata.Headers convertHeaders(Http2Headers http2Headers) {
     Metadata.Headers headers = new Metadata.Headers(convertHeadersToArray(http2Headers));
     if (http2Headers.authority() != null) {
-      headers.setAuthority(http2Headers.authority().toString());
+      headers.setAuthority(http2Headers.authority().toString(UTF_8));
     }
     if (http2Headers.path() != null) {
-      headers.setPath(http2Headers.path().toString());
+      headers.setPath(http2Headers.path().toString(UTF_8));
+    }
+    if (http2Headers.get(TIMEOUT) != null) {
+      headers.setTimeout(new Metadata.Timeout(http2Headers.get(TIMEOUT).toString(UTF_8)));
```

Comment:
Once you start using `Metadata.Key`, this could be removed as well (since `headers.setTimeout` would no longer exist). When we add server-side support, `ServerImpl` would use the `Metadata.Key` to parse the timeout and do its own processing.